examples/msbr/openmc_msbr_model.py

```diff
@@ -0,0 +1,681 @@
+        Name of extra cylindrical surface.
+
+    Returns
+    -------
+    gr_sq_neg : openmc.Intersection
+        The region bounding the outer surface of the graphite
+        element.
+    gr_extra_regions : list of (openmc.Region, str)
+        'Add-on' regions and their names for the graphite element.
+        Includes ribs, rib tips, and gap-filling regions.
+    inter_elem_channel : openmc.Region, list of (openmc.Region, str)
+        Inter-element channel region(s)
+    extra_surf : openmc.ZCylinder
+        Extra cylindrical surface used in the element.
+    """
+    # Square graphite element
+    gr_sq_neg = openmc.rectangular_prism(gr_sq_d*2,
+                                         gr_sq_d*2,
+                                         corner_radius=gr_sq_r)
+
+    # Rib tip surfaces
+    ul_t = openmc.ZCylinder(-l1, -l3, r_rib, name='rib_ul_tip')
+    br_t = openmc.ZCylinder(l1, l3, r_rib, name='rib_br_tip')
+    ru_t = openmc.ZCylinder(-l3, l1, r_rib, name='rib_ru_tip')
+    lb_t = openmc.ZCylinder(l3, -l1, r_rib, name='rib_lb_tip')
+
+    # Graphite element rib tip regions
+    rib_ul_t = -ul_t
+    rib_br_t = -br_t
+    rib_ru_t = -ru_t
+    rib_lb_t = -lb_t
+
+    if elem_type == 'core':
+        # Graphite element ribs for zones I-A and II-A
+        ul = openmc.ZCylinder(-l1, l2, r_rib, name='rib_ul')
+        br = openmc.ZCylinder(l1, -l2, r_rib, name='rib_br')
+        lb = openmc.ZCylinder(-l2, -l1, r_rib, name='rib_lb')
+        ru = openmc.ZCylinder(l2, l1, r_rib, name='rib_ru')
+
+        # Graphite element rib regions.
+        rib_ul = -ul
+        rib_br = -br
+        rib_lb = -lb
+        rib_ru = -ru
+
+        # inter-element fuel channel region
+        inter_elem_channel = +ul & +br & +lb & +ru
+
+    elif elem_type == 'cr':
+        # Parameters for control rod element
+        r_d = 1.16
+        e_d = 2 * r_d / np.sqrt(3)
+        r_c = 0.18
+
+        # Base rib region
+        ul = openmc.model.hexagonal_prism(origin=(-l1, l2), edge_length=e_d,
+                                          orientation='x', corner_radius=r_c)
+        br = openmc.model.hexagonal_prism(origin=(l1, -l2), edge_length=e_d,
+                                          orientation='x',corner_radius=r_c)
+        lb = openmc.model.hexagonal_prism(origin=(-l2, -l1), edge_length=e_d,
+                                          orientation='y',corner_radius=r_c)
+        ru = openmc.model.hexagonal_prism(origin=(l2, l1), edge_length=e_d,
+                                          orientation='y',corner_radius=r_c)
+
+        rib_ul = ul
+        rib_lb = br
+        rib_br = lb
+        rib_ru = ru
+
+        inter_elem_channel = ~ul & ~br & ~lb & ~ru
+
+    ribs = [[rib_ul, 'rib_ul'],
+            [rib_br, 'rib_br'],
+            [rib_ru, 'rib_ru'],
+            [rib_lb, 'rib_lb'],
+            [rib_ul_t, 'rib_ul_t'],
+            [rib_br_t, 'rib_br_t'],
+            [rib_ru_t, 'rib_ru_t'],
+            [rib_lb_t, 'rib_lb_t']]
+
```

Comment:
I thought your README said you wouldn't be including the ribs?  Or are these a different set of reflector ribs?

Reply:
These are the rip tips. In order to make a square lattice element, the tips of the ribs get cut off, so we need to add them back in at the position they would be in coming from adjacent elements. A picture of this is below.
![zone_ib_main_openmc](https://user-images.githubusercontent.com/45364492/202527215-ed63172f-020c-4e1d-b708-50528419b63e.png)